llvm/lib/Transforms/Yk/ShadowStack.cpp

```diff
@@ -62,171 +141,194 @@ class YkShadowStack : public ModulePass {
     return false;
   }
 
+  // Insert main's prologue.
+  //
+  // Main is a little different, in that it actually allocates the shadow stack
+  // and thus can use the allocation directly if it needs shadow space.
+  //
+  // Returns a pointer to the result of the call to malloc that was used to
+  // heap allocate memory for a shadow stack.
+  CallInst *insertMainPrologue(Function *Main, GlobalVariable *SSGlobal,
```

Comment:
I'm fine with this for now but I suspect the "right" long term fix will also fix threading. That probably means dynamically intercepting process / thread creation, rather than inserting code into `main` directly.